rustdoc: Fix errors in `components` and `ports`

komutumu opened the pull request:
- Corrected wrong spelling in `components/gfx/text/shaping/harfbuzz-rs`
- Corrected references in `components/layout_2020/fragment_tree`
- Corrected wrong spelling in  `ports/servoshell/minibrowser.rs`

---
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes are part of #31575 
- [X] These changes do not require tests because they don't change the functionality.

ktumuhairwe committed:
Fixed warnings in components/gfx/text/shaping/harfbuzz-rs  components/layout_2020/fragment_tree  ports/servoshell/minibrowser.rs

ktumuhairwe committed:
Fixed warnings in components/gfx/text/shaping/harfbuzz-rs  components/layout_2020/fragment_tree  ports/servoshell/minibrowser.rs.

atbrakhi commented on ports/servoshell/minibrowser.rs:
```diff
@@ -295,8 +295,8 @@ impl Minibrowser {
 
-    /// Updates the location field from the given [BrowserManager], unless the user has started
-    /// editing it without clicking Go, returning true iff it has changed (needing an egui update).
+    /// Updates the location field from the given [Minibrowser], unless the user has started
```
I think this should be `WebViewManager` instead of `BrowserManager`.  It was renamed in this [PR](https://github.com/servo/servo/pull/30842)

jahielkomu replied:
Thank you @atbrakhi for this review.
Let me correct them immediately.

jahielkomu replied:
Hey @atbrakhi changing it to ```WebViewManager``` causes a warning. 
I'm I missing something?

jahielkomu replied:
I was missing something, it was the spelling. 
Thank you @atbrakhi for the review, please kindly re-review and comment

atbrakhi commented on ports/servoshell/minibrowser.rs:
```diff
@@ -295,8 +295,8 @@ impl Minibrowser {
-    /// Updates the location field from the given [BrowserManager], unless the user has started
-    /// editing it without clicking Go, returning true iff it has changed (needing an egui update).
+    /// Updates the location field from the given [Minibrowser], unless the user has started
+    /// editing it without clicking Go, returning true if it has changed (needing an egui update).
```
this `iff` is correct, we don't need to change it to `if`. `iff` is shortened form of `if and only if`

komutumu committed:
Update minibrowser.rs

Returned "if" back to "iff"

komutumu committed:
Update minibrowser.rs

Changed ```BrowserManager``` to ```WebViewManager```

mrobinson commented on components/layout_2020/fragment_tree/fragment.rs:
```diff
@@ -30,7 +30,7 @@ pub(crate) enum Fragment {
     /// [BoxFragment] but it isn't positioned using normal in block flow
     /// positioning rules (margin collapse, etc). Instead, they are laid out by
-    /// the [SequentialLayoutState] of their float containing block formatting
+    /// the [PositioningFragment] of their float containing block formatting
```
`SequentialLayoutState` is correct here. I think you just need to import it.

jahielkomu replied:
Hello @mrobinson . Thank you for this review. 
I have imported ```SequentialLayoutState``` like this; ```use crate::flow::float::SequentialLayoutState; ``` I get an error message; ```warning: unused import: `crate::flow::float::SequentialLayoutState`
  --> components/layout_2020/fragment_tree/fragment.rs:26:5
   |
26 | use crate::flow::float::SequentialLayoutState;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default ```
   
   I'm missing the correct way to import? Kindly advise.

mrobinson replied:
Sorry, I may have mislead you here. I would try to remove the import and use ``crate::flow::float::SequentialLayoutState` directly in the rustdoc comment. If that doesn't work, I would replace the brackets with backticks instead.

komutumu committed:
Update fragment.rs

Changed ```[SequentialLayoutState]``` to ```[crate::flow::float::SequentialLayoutState]```

mrobinson reviewed (approved):
Thanks for the fixes!

Loirooriol commented on components/layout_2020/fragment_tree/fragment.rs:
```diff
@@ -30,7 +30,7 @@ pub(crate) enum Fragment {
     /// [BoxFragment] but it isn't positioned using normal in block flow
     /// positioning rules (margin collapse, etc). Instead, they are laid out by
-    /// the [SequentialLayoutState] of their float containing block formatting
+    /// the [crate::flow::float::SequentialLayoutState] of their float containing block formatting
     /// context.
```
Please balance these lines, the former is now considerably longer than the others in this paragraph, and the second has a single word.

jahielkomu replied:
I have updated the PR @Loirooriol 
Thank you.

komutumu committed:
Update fragment.rs

Balanced the length of paragraph lines in fragment.rs

Loirooriol reviewed (approved):
Thanks!

mrobinson committed:
Fix tidy errors
